### PR TITLE
Adds 'superset export_dashboards --dashboard-id/-i --dashboard-title/-t' options to CLI

### LINKED ISSUE #7444: Add CLI options to export individual dashboards by id/title
**Is your feature request related to a problem? Please describe.**

I need to be able to export one dashboard at a time.

**Describe the solution you'd like**

```bash
rjurney$ superset export_dashboards --help

Usage: superset export_dashboards [OPTIONS]

  Export dashboards to JSON

Options:
  -f, --dashboard-file TEXT   Specify the the file to export to
  -p, --print_stdout          Print JSON to stdout
  -i, --dashboard-id INTEGER  Specify dashboard id to export
  -t, --dashboard-title TEXT  Specify dashboard title to export
  --help                      Show this message and exit.

```

**Describe alternatives you've considered**

Manually filtering the JSON is annoying.

### PR DESCRIPTION
**Pull request by rjurney**
Added a couple of options to the CLI for convenience for exporting a single dashboard at a time by ID or title.

See #7444.

### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Added `--dashboard-id/-i` and `--dashboard-title/-t` options to `superset export_dashboards` CLI command.

Added these as optional parameters to `superset.utils.dashboard_import_export.export_dashboards`:

```python
def export_dashboards(session, dashboard_id=None, dashboard_title=None):
```
which is only called by the CLI.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before help UI:

```bash
rjurney$ superset export_dashboards --help

Usage: superset export_dashboards [OPTIONS]

  Export dashboards to JSON

Options:
  -f, --dashboard-file TEXT   Specify the the file to export to
  -p, --print_stdout          Print JSON to stdout
  --help                      Show this message and exit.
```

After help UI:

```bash
rjurney$ superset export_dashboards --help

Usage: superset export_dashboards [OPTIONS]

  Export dashboards to JSON

Options:
  -f, --dashboard-file TEXT   Specify the the file to export to
  -p, --print_stdout          Print JSON to stdout
  -i, --dashboard-id INTEGER  Specify dashboard id to export
  -t, --dashboard-title TEXT  Specify dashboard title to export
  --help                      Show this message and exit.
```

### TEST PLAN

* `tox -e py36 tests/import_export_tests.py` passes
* manual testing works

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #7444
- [x] Changes UI (CLI)
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

@mistercrunch

### CODE REVIEW

File: superset/cli.py [hunk 1]
```diff
@@ -187,9 +187,18 @@ def import_dashboards(path, recursive):
 @click.option(
     '--print_stdout', '-p', is_flag=True, default=False,
     help='Print JSON to stdout')
-def export_dashboards(print_stdout, dashboard_file):
+@click.option(
```
Comment (mistercrunch) [inline superset/cli.py]: Could be nice to support multiple ids here:
https://click.palletsprojects.com/en/5.x/options/#multiple-options

File: superset/utils/dashboard_import_export.py [hunk 2]
```diff
@@ -38,12 +38,17 @@ def import_dashboards(session, data_stream, import_time=None):
...
-def export_dashboards(session):
+def export_dashboards(session, dashboard_id=None, dashboard_title=None):
     """Returns all dashboards metadata as a json dump"""
     logging.info('Starting export')
     dashboards = session.query(Dashboard)
     dashboard_ids = []
     for dashboard in dashboards:
-        dashboard_ids.append(dashboard.id)
-    data = Dashboard.export_dashboards(dashboard_ids)
+        if dashboard_id or dashboard_title:
```
Comment (mistercrunch) [inline superset/utils/dashboard_import_export.py]: In envs with lots of dashboards, it'd be more efficient to add a query filter, as in:

[untested]
```python
query = session.query(Dashboard)
if dashboard_ids:
    query = query.filter(Dasbhoard.id.in_(dashboard_ids))
if dashboard_titles:
    query = query.filter(Dasbhoard.title.in_(dashboard_titles))
```

File: superset/utils/dashboard_import_export.py [hunk 3]
```diff
@@ -38,12 +38,17 @@ def import_dashboards(session, data_stream, import_time=None):
...
+            if dashboard.id == dashboard_id or \
+                dashboard.dashboard_title == dashboard_title:
+                dashboard_ids.append(dashboard.id)
+        else:
+            dashboard_ids.append(dashboard.id)
+    data = Dashboard.export_dashboards(dashboard_ids) if dashboard_ids else {}
```
Comment (mistercrunch) [inline superset/utils/dashboard_import_export.py]: could be good to:
```python
dashboard_ids = [d.id for d in query.all()]
if not dashboard_ids:
    click.secho('No dashboard found', fg='red)
    sys.exit(1)
```

File: superset/utils/dashboard_import_export.py (hunk 2 above)
Comment (rjurney) [reply superset/utils/dashboard_import_export.py]: Good idea.

File: superset/utils/dashboard_import_export.py (hunk 3 above)
Comment (rjurney) [reply superset/utils/dashboard_import_export.py]: I've made it work this way and have added a `superset.exceptions.DashboardNotFoundException` that `dashboard_import_export.export_dashboards()` raises and the `cli.py` caller intercepts so he can use `click.echo(click.style(message, fg='red'))`. It does not warn about a single incorrect id/title among multiple id/title arguments. I could add that as well if you would like.

File: superset/config.py [hunk 4]
```diff
@@ -608,6 +608,8 @@ class CeleryConfig(object):
 # localtime (in the tz where the superset webserver is running)
 IS_EPOCH_S_TRULY_UTC = False
 
+# Default dashboard export directory
+DASHBOARD_EXPORT_DIR = '/tmp/'
```
Comment (mistercrunch) [inline superset/config.py]: Unused?

File: superset/utils/dashboard_import_export.py [hunk 5]
```diff
@@ -38,12 +40,23 @@ def import_dashboards(session, data_stream, import_time=None):
...
+                             Dashboard.dashboard_title.in_(dashboard_titles))
+
+    export_dashboard_ids = [d.id for d in query.all()]
+
+    data = {}
+    if not export_dashboard_ids:
+        logging.error('No dashboards found!')
+        raise DashboardNotFoundException('No dashboards found!')
+    else:
+        data = Dashboard.export_dashboards(export_dashboard_ids) if export_dashboard_ids else {}
```
Comment (mistercrunch) [inline superset/utils/dashboard_import_export.py]: No need for the ternary `if` here anymore I think

File: superset/config.py (hunk 4 above)
Comment (rjurney) [reply superset/config.py]: Truth, bled over from additional example work. Fixed.

File: superset/utils/dashboard_import_export.py (hunk 5 above)
Comment (rjurney) [reply superset/utils/dashboard_import_export.py]: k

File: superset/utils/dashboard_import_export.py (hunk 2 above)
Comment (rjurney) [reply superset/utils/dashboard_import_export.py]: Ended up doing an OR between them in a single query.

**Comment by mistercrunch** [comment]
Build has flake8 issues 
https://travis-ci.org/apache/incubator-superset/jobs/529873946

**Comment by rjurney** [comment]
I'm thinking of forgetting all about this pull request because it is reproduced in and superseded by the work on redoing the examples stuff.

**Pull request closed without merging**